### ISSUE TITLE
Generate a Nuget.Config file by default, don't require users to create it

### ISSUE THREAD

**Issue by atrauzzi** [aspnet/Home#903]
Reading through the instructions for getting started on Linux, I'm [being told to create `Nuget.Config`](http://docs.asp.net/en/latest/getting-started/installing-on-linux.html#add-nuget-sources), a fairly obscure file that doesn't seem like it would ever change for the average user.

This step could be completely eliminated by just either having `dnvm` use the correct sources by default, or by generating this file during any initial operation.

The beauty of this is that people who want to use dnx won't be plagued by these very specific non-development considerations.  Things should _just work_, similar to how npm, composer, gem etc, where they don't require you to configure repositories.

**Comment by davidfowl** [comment]
Visual Studio does this, I think the yeomen generator does it as well. Who should generate the config file?

**Comment by davidfowl** [comment]
> This step could be completely eliminated by just either having dnvm use the correct sources by default, or by generating this file during any initial operation.

Doesn't make sense in dnvm. That's the wrong tool. dnu does make a default global NuGet.config on first run so I'm not 100% sure what you mean.

> The beauty of this is that people who want to use dnx won't be plagued by these very specific non-development considerations. Things should just work, similar to how npm, composer, gem etc, where they don't require you to configure repositories.


Sure, if everything you need is on nuget.org. People using nuget have custom feeds and if you want access to ci builds you also need to configure feeds.

**Comment by atrauzzi** [comment]
@davidfowl So `Nuget.Config` doesn't need to be created?

**Comment by davidfowl** [comment]
I'd recommend having one in your project template by default which is what we do. But if none exists, one is created machine wide the very first time you use nuget or dnu.

**Comment by atrauzzi** [comment]
Extra configuration files are going to add up and are a very prominent reason why pre-aspnet-5 projects are so challenging to manage.  XML this, JSON that, this in your home dir, this in your system, that in your project...

If developers could stick to the defaults and deviate when needed, it would certainly help.  Especially hidden config dir structures which are always going to be arcane and 99% of the time should be managed by tools, not by hand.

All this is to say:  For the sake of that documentation, if the `Nuget.Config` file is not needed, it would be worthwhile to remove that step from the "Installing ASP.NET" document.  Considering by this point, no project has even been made yet.

**Comment by MisinformedDNA** [comment]
I used VSTS in March and I swear I just added the NuGet build task and everything just worked. Now, on a new project (and new company) it complains that I need a nuget.config. I am currently only using nuget.org, so I don't need anything customized. So all .NET code should assume a default nuget.config unless one is provided. If you insist that one must be provided, provide us with a simple file upon project creation in the solution or at least create a file template that has the basics for customization.
